libraries/mongodb.rb

```diff
@@ -39,7 +48,14 @@ def self.configure_replicaset(node, name, members)
     begin
       connection = nil
       rescue_connection_failure do
-        connection = Mongo::Connection.new('localhost', node['mongodb']['config']['port'], :op_timeout => 5, :slave_ok => true)
+        connection = self.get_connection(node)
+        if node['mongodb']['config']['auth'] == true
+          begin
+            connection.add_auth("admin", node['mongodb']['admin']['username'], node['mongodb']['admin']['password'])
+          rescue Mongo::AuthenticationError => e
+            Chef::Log.warn("Unable to authenticate as admin user. If this is a fresh install, ignore warning: #{e}")
```

Comment:
You're proposing a node.set["mongodb"]["admin"]["created"] = true
After creation in user_management.rb (recipes)?